docs/coding-style.md

```diff
@@ -45,23 +45,38 @@ Variable name `sut` (System Under Test) is recommended in unit tests that really
 * Dot before an invocation `.Method()` is placed at the beginning of a line.
 * The comma separating arguments is placed at the end of a line.
 * Method declaration parameters should be on the same line. If S103 is violated, parameters should be placed each on a separate line; the first parameter should be on the same line with the declaration; the other parameters should be aligned with the first parameter.
-```
-public void MethodWithManyParameters(int firstParameter,
-                                     string secondParameter,
-                                     Function<int, string, string> complexParameter);
-```
+    ```
+    public void MethodWithManyParameters(int firstParameter,
+                                         string secondParameter,
+                                         Function<int, string, string> complexParameter);
+    ```
+* Chained invocations and member accesses violating S103 can have a chain on the first line until the first lambda, every other `.Invocation()` or `.Member` should be on a separate line, aligned with a left-most single indendation.
+    ```
+    object.Property.Method().Select(x => x.Something)
```

Comment:
In my opinion, all method calls should be on separate lines. I prefer not to have this kind of exception ("until the first lambda").
Our conventions should be intuitive and easy to follow.

Reply:
That's why there's `can`, I wanted to define lambda as the starting point for the mandatory split. Do you have another idea how to allow boring introductions like 
```
c.GetSyntaxTree().GetRoot().DescendantNodes().OfType<xxx>()
    .Where(xxx)
```

Reply:
I think is an enormous waste of (screen)space. It also is likely to be violated (unintentionally) when renaming something.

I would argue that this is a preferable way to allign this.

``` C#
public void MethodWithManyParameters(
  int firstParameter,
  string secondParameter,
  Function<int, string, string> complexParameter);
```
It is unlikely not to fit in the screen, it is not dependent on the length of any name, and easy to read. (Just my 2cts)

Reply:
Hi @Corniel, we were discussing here chain invocations. For method declaration parameters we already defined a rule very close to your example.

@pavel-mikula-sonarsource : I think we should be consistent and have all method calls on separate lines. 

@andrei-epure-sonarsource , @csaba-sagi-sonarsource, @mary-georgiou-sonarsource: WDYT?

Reply:
I would say this is something that should be discussed during the How we work meeting, so everybody is aligned and hopefully happy with the decision we take. I don't see why the urgency to make the change in the coding style, we will have the meeting anyway in 3 days time.

Reply:
I agree with having chain of property access on the same line and method calls on separate lines.

The example @pavel-mikula-sonarsource gives is getters which are like properties (you can easily see them as `c.SyntaxTree.Root.DescendantNodes`), probably the MS Team made them methods because of their perf? And from this point of view, having them on separate lines will outline there is a higher cost associated with them (even if they're boring).

For example, `DescendantNodes()` without a filter can lead to major perf hit (e.g. https://github.com/SonarSource/sonar-security/pull/1783/files).

In short, I favor methods on separate lines because they do an action which has a cost.

And we can discuss this in 3 days.

Reply:
One more point to discuss on the following "How we work" meeting:
        
 ```csharp
private readonly VerifierBuilder builderCS = new VerifierBuilder()
        .AddAnalyzer(() => new())
        .WithBasePath(@"Hotspots")
        .AddReferences(GetAdditionalReferences());
```
or

 ```csharp
private readonly VerifierBuilder builderCS = new VerifierBuilder()
                                                                      .AddAnalyzer(() => new())
                                                                      .WithBasePath(@"Hotspots")
                                                                      .AddReferences(GetAdditionalReferences());
```
so that the invocations are closer to the invocation target.

Reply:
@mary-georgiou-sonarsource I'd go for the first as well, for simplicity and having a single rule without too many exceptions